controllers/podhttpchaos/types.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podhttpchaos
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/utils"
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+)
+
+// Handler applys podiochaos
+type Handler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// Apply flushes io configuration on pod
+func (h *Handler) Apply(ctx context.Context, chaos *v1alpha1.PodHttpChaos) error {
+	h.Log.Info("updating http chaos", "pod", chaos.Namespace+"/"+chaos.Name, "spec", chaos.Spec)
+
+	pod := &v1.Pod{}
+
+	err := h.Client.Get(ctx, types.NamespacedName{
+		Name:      chaos.Name,
+		Namespace: chaos.Namespace,
+	}, pod)
+	if err != nil {
+		h.Log.Error(err, "fail to find pod")
+		return err
+	}
+
+	pbClient, err := utils.NewChaosDaemonClient(ctx, h.Client, pod)
+	if err != nil {
+		return err
+	}
+	defer pbClient.Close()
+
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return fmt.Errorf("%s %s can't get the state of container", pod.Namespace, pod.Name)
+	}
+
+	containerID := pod.Status.ContainerStatuses[0].ContainerID
+	if chaos.Spec.Container != nil &&
+		len(strings.TrimSpace(*chaos.Spec.Container)) != 0 {
+		containerID = ""
+		for _, container := range pod.Status.ContainerStatuses {
+			if container.Name == *chaos.Spec.Container {
```

Comment:
It's impossible to support multiple containers in podhttpchaos, as all containers inside a pod share the same network namespace.

Reply:
Removed.